src/main/java/reposense/RepoSense.java

```diff
@@ -93,17 +94,23 @@ public static void main(String[] args) {
         try {
             authorConfigs = new AuthorConfigCsvParser(cliArguments.getAuthorConfigFilePath()).parse();
             RepoConfiguration.merge(repoConfigs, authorConfigs);
-        } catch (IOException ioe) {
-            // IOException thrown as author-config.csv is not found.
+        } catch (FileNotFoundException fnfe) {
```

Comment:
@jylee-git, @fzdy1914, I added a `catch` block here so we can warn the user if the author / group config files have an invalid header size (rather than silently ignoring the problem).

Should this be done within this PR, or separately?

Reply:
> Throw InvalidCsvException out from parse in CsvParser, and use it to log a warning for author-config.csv and group-config.csv, in the driver RepoSense.java code.

Hi, can you consider my Logic? Since we are dealing with CSV files, how about we directly set the `InvalidCsvException` a child of `IOException`. Then, instead of catching `FileNotFoundException`, we catch `InvalidCsvException` here and do the warning. We still leave the other `IOException` unhandled. In the meantime, we do not need to tough the test case a lot as it is a child of `IOException`.

Reply:
But then we will be silently ignoring the IOExceptions besides the InvalidCsvException, right?
Would it be better to warn user about all IOExceptions that occur while parsing author and group configs, so the user can re-run the analysis ?

Reply:
OK, let's ask the senior's opinion. @yong24s @eugenepeh Your inputs, please?

Reply:
why is `FileNotFoundException` used for `if the author / group config files have an invalid header size`?

Reply:
His implementation is only ignoring `FileNotFoundException` instead of ignoring all `IOException` as we previously did.

Reply:
@eugenepeh,  `FileNotFoundException` **will not** be thrown if the config files have an invalid header size. The `parse` method in `CsvParser` will throw an `IOException` if the config files have an invalid header size.

In our current implementation, we are silently ignoring all `IOExceptions` which are thrown while parsing author / group config files. However, it would be better to alert the user if the author / group configs have an invalid header size (or any other problem with the file).

Hence, I decided to distinguish between the case where the author / group config **is not present** and the case where the author / group config **is present but cannot be parsed because they are malformed / other IO error.**

The `FileNotFound` catch-block covers the first case (author / group config **is not present**).

Reply:
I see, sounds good to me